comp/core/flare/flare_test.go

```diff
@@ -26,6 +28,11 @@ func TestFlareCreation(t *testing.T) {
 			t,
 			logimpl.MockModule(),
 			config.MockModule(),
+			secretsimpl.MockModule(),
+			fx.Provide(func(secretMock secrets.Mock) secrets.Component {
```

Comment:
Can you move this code inside `MockModule()` like [here](https://github.com/DataDog/datadog-agent/pull/23388/files#diff-8d4ced3b62f2fe71f1563e062477d5b2b35c9aa91bcf13136cf0e513d36f2a9bR47) so if there is another part of the code that use `Mock` it will not be needed to have these lines of code?